legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-bigquery/src/main/java/org/finos/legend/engine/persistence/components/relational/bigquery/sql/BigQueryDataTypeToLogicalDataTypeMapping.java

```diff
@@ -0,0 +1,71 @@
+// Copyright 2023 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.engine.persistence.components.relational.bigquery.sql;
+
+import org.finos.legend.engine.persistence.components.logicalplan.datasets.DataType;
+import org.finos.legend.engine.persistence.components.logicalplan.datasets.FieldType;
+import org.finos.legend.engine.persistence.components.relational.sql.JdbcPropertiesToLogicalDataTypeMapping;
+
+public class BigQueryDataTypeToLogicalDataTypeMapping implements JdbcPropertiesToLogicalDataTypeMapping
+{
+    private static final String INT64 = "INT64";
+    private static final String INTEGER = "INTEGER";
+    private static final String NUMERIC = "NUMERIC";
+    private static final String FLOAT64 = "FLOAT64";
+    private static final String FLOAT = "FLOAT";
+    private static final String STRING = "STRING";
+    private static final String BYTES = "BYTES";
+    private static final String BOOL = "BOOL";
+    private static final String BOOLEAN = "BOOLEAN";
+    private static final String DATE = "DATE";
+    private static final String TIME = "TIME";
+    private static final String DATETIME = "DATETIME";
+    private static final String TIMESTAMP = "TIMESTAMP";
+    private static final String JSON = "JSON";
+
+    public FieldType getDataType(String typeName, String dataType, Integer columnSize, Integer decimalDigits)
+    {
+        switch (typeName)
+        {
+            case INT64:
+            case INTEGER:
+                return FieldType.builder().dataType(DataType.INTEGER).build();
+            case NUMERIC:
+                return FieldType.builder().dataType(DataType.NUMERIC).length(columnSize).scale(decimalDigits).build();
+            case FLOAT64:
+            case FLOAT:
+                return FieldType.builder().dataType(DataType.FLOAT).build();
```

Comment:
On the same line as prev comment on INT64, FLOAT and FLAOT 64 are two different types that have different ranges of values they can store.

Reply:
BigQuery only supports FLOAT64, so we have mapped both these logical types to FLOAT64